### PR TITLE
Fix(T33137): accessible notifications

### PR DESCRIPTION
**Pull request by sakutademos**
**Ticket:** https://yaits.demos-deutschland.de/T33137

**Description:** This PR fixes the issue of duplicated invalid fields and impoves accessibility in the DpValidation action by including invalid fields into the error message, if they exist.

**Linked PRs**
[PR](https://github.com/demos-europe/demosplan-core/pull/2251) in demosplan

### CODE REVIEW

**Review by elmasdemos — APPROVED** [review]
lgtm, code review

**Review by elmasdemos — APPROVED** [review]
lgtm, code review only

File: src/lib/validation/utils/assignHandlerForTrigger.js [hunk 1]
```diff
@@ -44,10 +45,19 @@ export default function assignHandlerForTrigger (triggerButton, form) {
       invalidCustomErrorFields.forEach(field => {
         dplan.notify.notify('error', Translator.trans(field.validationMessage))
       })
-      // Make sure the default notification is displayed as needed
+      const nonEmptyFieldNames = [...new Set(
+        validatedForm.invalidFields
+          .map(field => field.getAttribute('data-dp-validate-error-fieldname'))
+          .filter(Boolean)
```
Comment (salisdemos) [inline src/lib/validation/utils/assignHandlerForTrigger.js:51]: what does this do? Looks for me, as if it always return true

File: src/lib/validation/utils/assignHandlerForTrigger.js [hunk 2]
```diff
@@ -44,10 +45,19 @@ export default function assignHandlerForTrigger (triggerButton, form) {
...
+      )]
+
       if (invalidCustomErrorFields.length === 0 || invalidCustomErrorFields.length < validatedForm.invalidFields) {
-        dplan.notify.notify('error', Translator.trans('error.mandatoryfields.no_asterisk'))
+        const fieldsString = nonEmptyFieldNames ? nonEmptyFieldNames.join(', ') : ' '
+        const errorMandatoryFields = de.error.mandatoryFields.intro + fieldsString + de.error.mandatoryFields.outro
```
Comment (salisdemos) [inline src/lib/validation/utils/assignHandlerForTrigger.js:56]: Are the trans-keys defined somewhere? Can't find them. (only another usage, but no declaration)
```suggestion
        const errorMandatoryFields = `${de.error.mandatoryFields.intro} ${fieldsString} ${de.error.mandatoryFields.outro}`
```

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: src/lib/validation/utils/assignHandlerForTrigger.js (hunk 2 above)
Comment (sakutademos) [reply src/lib/validation/utils/assignHandlerForTrigger.js:56]: yes, they are defined here --> '~/components/shared/translations'

Comment (salisdemos) [reply src/lib/validation/utils/assignHandlerForTrigger.js:56]: sorry. I am still used to find transkeys when searching for the whole string.

**Review by salisdemos — APPROVED** [review]

File: src/lib/validation/utils/assignHandlerForTrigger.js (hunk 1 above)
Comment (sakutademos) [reply src/lib/validation/utils/assignHandlerForTrigger.js:51]: it return a list of fields. It extracts the value of the `data-dp-validate-error-fieldname` attribute from each element and filter empty/falsy values from the array.

Comment (salisdemos) [reply src/lib/validation/utils/assignHandlerForTrigger.js:51]: Didn't now that https://stackoverflow.com/questions/30016773/javascript-filter-true-booleans

**Pull request merged**